src/SurveyService.Validation/Question/CreateQuestionRequestValidator.cs

```diff
@@ -0,0 +1,51 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.SurveyService.Data.Interfaces;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Question;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Question.Filters;
+using LT.DigitalOffice.SurveyService.Validation.Question.Interfaces;
+using System;
+
+namespace LT.DigitalOffice.SurveyService.Validation.Question;
+
+public class CreateQuestionRequestValidator : AbstractValidator<CreateSingleQuestionRequest>, ICreateQuestionRequestValidator
+{
+  public CreateQuestionRequestValidator(
+    IQuestionRepository _questionRepository
+    )
+  {
+    RuleFor(q => q.Content)
+      .MaximumLength(300)
+      .WithMessage("Content is too long.");
+
+    When(q => q.Deadline.HasValue, () =>
+    {
+      RuleFor(q => q.Deadline)
+        .Must(d => d > DateTime.UtcNow.AddDays(1).AddSeconds(2))
+        .WithMessage("The deadline must be at least 24 hours from now.");
+    });
+
+    When(q => q.GroupId.HasValue, () =>
+    {
+      RuleFor(q => q)
+        .MustAsync(async (q, _) => (await _questionRepository.GetPropertiesAsync(new GetQuestionPropertiesFilter() { GroupId = q.GroupId})).Deadline == q.Deadline)
```

Comment:
We'll get NullReferenceException if we don't have such GroupId 
(Or we sure that questions are created immediately after the groups?)

Reply:
Here we add a question to a previously created group, and there is 100% already at least one question in that group. If it happens and there is no question for some reason - GetProperties returns an empty DbQuestion and the exception will not throw, but validator return error

Reply:
Ok)